client/rpc/block_converter.go

```diff
@@ -0,0 +1,223 @@
+package rpc
+
+import (
+	"encoding/json"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	ctypes "github.com/tendermint/tendermint/rpc/core/types"
+	"github.com/tendermint/tendermint/types"
+	"github.com/tendermint/tendermint/version"
+)
+
+// ResultBlock represents a single block with metadata
+type ResultBlock struct {
+	BlockMeta BlockMeta `json:"block_meta"`
+	Block     Block     `json:"block"`
+}
+
+// BlockMeta contains meta information about a block - namely, it's ID and Header.
+type BlockMeta struct {
+	BlockID types.BlockID `json:"block_id"` // the block hash and partsethash
+	Header  Header        `json:"header"`   // The block's Header
+}
+
+// Header defines a wrapper around Tendermint's Header type overriding various fields.
+// nolint: structtag
+type Header struct {
+	// embed original type
+	types.Header
+
+	// override fields from original type
+	Version         Consensus      `json:"version"`
+	Height          int64          `json:"height,string"`    // Override int64 fields do to json.Marshal not converting them to string
```

Comment:
Instead of copying the side comment, I would just add a single comment to the top and group them:


```go
type Header struct {
  // override fields so json.Marshal will marshal in accordance with amino JSON format
  Version         Consensus
  Height          int64          `json:"height,string"`
  // ...

  ProposerAddress sdk.ValAddress `json:"proposer_address"`
}